src/parser/permit-generation-module.ts

```diff
@@ -36,42 +33,37 @@ interface Payload {
   issue: { node_id: string };
 }
 
-export class PermitGenerationModule implements Module {
-  readonly _configuration: PermitGenerationConfiguration | null = configuration.incentives.permitGeneration;
-  readonly _supabase = createClient<Database>(process.env.SUPABASE_URL, process.env.SUPABASE_KEY);
+export class PermitGenerationModule extends BaseModule {
+  readonly _configuration: PermitGenerationConfiguration | null = this.context.config.incentives.permitGeneration;
+  readonly _supabase = createClient<Database>(this.context.env.SUPABASE_URL, this.context.env.SUPABASE_KEY);
 
   async transform(data: Readonly<IssueActivity>, result: Result): Promise<Result> {
     const payload: Context["payload"] & Payload = {
       ...context.payload.inputs,
-      issueUrl: program.eventPayload.issue.html_url,
-      evmPrivateEncrypted: configuration.evmPrivateEncrypted,
-      evmNetworkId: configuration.evmNetworkId,
-      erc20RewardToken: configuration.erc20RewardToken,
+      issueUrl: this.context.payload.issue.html_url,
+      evmPrivateEncrypted: this.context.config.evmPrivateEncrypted,
+      evmNetworkId: this.context.config.evmNetworkId,
+      erc20RewardToken: this.context.config.erc20RewardToken,
     };
     const issueId = Number(payload.issueUrl.match(/\d+$/)?.[0]);
     payload.issue = {
-      node_id: program.eventPayload.issue.node_id,
+      node_id: this.context.payload.issue.node_id,
     };
-    const env = Value.Default(envConfigSchema, process.env) as EnvConfigType;
-    if (!envValidator.test(env)) {
-      console.warn("[PermitGenerationModule] Invalid env detected, skipping.");
-      for (const error of envValidator.errors(env)) {
-        console.error(error);
-      }
-      return Promise.resolve(result);
-    }
+    const env = this.context.env;
     const isPrivateKeyAllowed = await this._isPrivateKeyAllowed(
       payload.evmPrivateEncrypted,
-      program.eventPayload.repository.owner.id,
-      program.eventPayload.repository.id,
+      this.context.payload.repository.owner.id,
+      this.context.payload.repository.id,
       env
     );
     if (!isPrivateKeyAllowed) {
-      console.warn("[PermitGenerationModule] Private key is not allowed to be used in this organization/repository.");
+      this.context.logger.error(
+        "[PermitGenerationModule] Private key is not allowed to be used in this organization/repository."
+      );
       return Promise.resolve(result);
     }
     const eventName = context.eventName as SupportedEvents;
-    const octokit = getOctokitInstance();
+    const octokit = this.context.octokit as unknown as Octokit;
```

Comment:
Do you think the SDK should export a function to create the `customOctokit` it uses to help avoid casts?

It won't fix everything though because `permit-generation` is running off of it's own `Context` and `octokit` type so it'll need bumped too.

A dedicated task for it for sure

Reply:
Yes the only reason is the mismatched types because `permit-generation` uses `Octokit core` and here we use `Octokit rest` with different plugins.